src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.List;
+
+/**
+ * The main class of Mitsuki the Chat bot.
+ *
+ * @author Tee Chu jie
+ */
 public class Duke {
+    /**
+     * A list of commands that can be given to Mitsuki.
+     */
+     static ArrayList<String> commands = new ArrayList<>();
+
+    /**
+     * Initiating the scanner for user input.
+     */
+    static Scanner scan = new Scanner(System.in);
+
+    /**
+     * Initiating the user's todo list.
+     */
+    static ArrayList<Task> list = new ArrayList<>();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // Greeting the user.
+        System.out.println("Hello! My name is Mitsuki, nice to meet you!\n"
+                + "What can I do for you today?\n");
+
+        try {
+            list = get("list.txt");
+            System.out.println("I have loaded your previously saved list for you. :)");
+        } catch(FileNotFoundException ex) {
+            System.out.println("No saved list found. A new list will be used!");
+            File saveFile = new File("list.txt");
+            try {
+                saveFile.createNewFile();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        // Adding the commands to the list of commands Mitsuki can execute.
+        commands.add("help");
+        commands.add("list");
+        commands.add("todo");
+        commands.add("deadline");
+        commands.add("event");
+        commands.add("mark");
+        commands.add("unmark");
+        commands.add("delete");
+        commands.add("bye");
+
+
+
+        // Initiating the 'command' variable.
+        String command = "nil";
+
+        // The loop that takes in user input and determine what to do.
+        while (!command.equals(null)) {
+            // Taking in next user input.
+            command = scan.next();
+
+            // Making sure user is giving a valid command.
+            try {
+                DukeException.validate(command);
+            } catch(DukeException ex) {
+                System.out.println("Sorry, I am unable to do that at the current moment.");
+                System.out.println("Please type 'help' for a list of commands you can give me! :)");
+            }
```

Comment:
Perhaps using the `default` clause in the switch statement below to handle invalid commands would be intuitive here?